packages/cli/src/foundry.ts

```diff
@@ -153,3 +161,40 @@ export async function getFoundryArtifact(name: string, baseDir = '', includeSour
     linkReferences: artifact.bytecode.linkReferences,
   };
 }
+
+
+export async function forgeBuild(options: Record<string, any>, cannonfile: string) {
+
+  const cannonfilePath = path.resolve(cannonfile);
+  const projectDirectory = path.dirname(cannonfilePath);
+
+log(bold('Building the foundry project...'));
+if (!options.skipCompile) {
+  // use --build-info to output build info
+  // ref: https://github.com/foundry-rs/foundry/pull/7197
+  const forgeBuildArgs = [
+    'build',
+    '--build-info',
+    '--ast',
+    ...fromFoundryOptionsToArgs(pickForgeBuildOptions(options), forgeBuildOptions),
+  ];
+
+  const forgeBuildProcess = spawn('forge', forgeBuildArgs, { cwd: projectDirectory, shell: true });
+
+  await new Promise((resolve, reject) => {
+    forgeBuildProcess.on('exit', (code) => {
+      if (code === 0) {
+        log(gray('forge build succeeded'));
+      } else {
+        log(red('forge build failed'));
+        log(red('Make sure "forge build" runs successfully or use the --skip-compile flag.'));
+        return reject(new Error(`forge build failed with exit code "${code}"`));
+      }
+
+      resolve(null);
+    });
+  });
+} else {
+  log(yellow('Skipping forge build...'));
```

Comment:
Here could be something like: `log(yellow('--no-build flag detected, skipping forge build...'));`?🤔